### ISSUE TITLE
How to set the points of the clipped surface

### ISSUE THREAD

**Issue by Nowfal1993** [pyvista/pyvista-support#119]
## Description
for example, I have a random surface like this 
![grafik](https://user-images.githubusercontent.com/59694177/73665373-45073280-46a1-11ea-82f0-067aba24e706.png)
if I am clipping the surface with a plane along the z axis. The results would be like this:
![grafik](https://user-images.githubusercontent.com/59694177/73665515-85ff4700-46a1-11ea-8c82-5ce6133a79fb.png)
But how we can set the points of the clipped surface to the points of the surface(in this case the points on the plane along the z axis) with which i clipped the real surface, so that I can avoid the empty spaces in the final surface. Thanks in advance

**Comment by banesullivan** [comment]
To clarify, are you trying to fill the "holes" resulting from the clip? 

When clipping 3D cells, intersection polygons are created along the clipping plane but with 2D cells like in your example, there is no way to safely "fill" the resulting "holes" that you see there.

**Comment by Nowfal1993** [comment]
Hi, Actually not trying filling the holes. If I clip the surface at z = 10(say). Instead of void points at the clipped portion, just wanna give the value z = 10.

**Comment by banesullivan** [comment]
So you want to use the plane to sort of “squish” the surface down?

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by GuillaumeFavelier** [comment]
If your function is always an elevation, what you ask is the same as clamping the Z value at 10 and clipping is not necessary.

**Comment by Nowfal1993** [comment]
> 
> 
> So you want to use the plane to sort of “flatten” the surface down?
> 
> Sent with [GitHawk](http://githawk.com)

yes. but not below the plane. only points above the plane.

**Comment by Nowfal1993** [comment]
> 
> 
> If your function is always an elevation, what you ask is the same as [clamping](https://en.wikipedia.org/wiki/Clamping_(graphics)) the Z coordinates at 10 and clipping is not necessary.

Yes we can do it manaully, But my prior funtion is in such a way that its return a normal and centroid of the plane.  But if the plane normal is not in eaxct z direction,, In such cases I cannot find the clipping ponts separtely. but I can clip the surface with the plane with certain centroid and the normal.

**Comment by GuillaumeFavelier** [comment]
I think I do and in my opinioin, it looks close to [topological simplification](https://topology-tool-kit.github.io/persistentHomologyDummies.html):

![image](https://user-images.githubusercontent.com/18143289/73671863-25293c00-46ac-11ea-928c-712e12a479e2.png)
*Image coming from a different project called [TopologyToolkit](https://topology-tool-kit.github.io/gallery.html)*
